student_explorer/cron.py

```diff
@@ -0,0 +1,191 @@
+        user_json = json.loads(user_result.text)
+        return user_json.get("id", "")
+
+    def mentor_in_affiliated_department(self, mentor_uniqname):
+        """
+        checks whether the mentor is in affiliated department
+        :param mentor_uniqname:
+        :return:
+        """
+        in_affiliated_department = False
+
+        # get the array of affiliated department from config settings
+        deptAffiliations = config("DEPT_AFFILIATION", default="").split(",")
+
+        logger.info("mentor uniquename=" + mentor_uniqname)
+        resp = apiUtil.api_call(f"MCommunity/People/{mentor_uniqname}", "mcommunity")
+        person_json = json.loads(resp.text)
+        if (person_json.get("httpCode", "") =="401"):
+            # if unauthorized, respose text will be:
+            # { "httpCode":"401", "httpMessage":"Unauthorized", "moreInformation":"Client id not registered." }
+            logger.warn(f"Not authorized to call MCommunity API")
+        elif (person_json.get("person", "").get("errors") !=""):
+            # cannot find user
+            logger.warn(f"Cannot find user {mentor_uniqname} in MCommunity")
+        else:
+            advisor_json = json.loads(resp.text)
+            user_affiliation = advisor_json.get("person").get("affiliation", "")
+            # array of affiliated department.
+            # The advisors of those department will be added as observers to student's Canvas courses
+            for dept in deptAffiliations:
+                if dept in user_affiliation:
+                    # advisor with right dept affiliation
+                    in_affiliated_department = True
+
+        return in_affiliated_department
+
+    def add_user_as_observer_to_course(self, class_site_id, mentor_canvas_user_id, user_canvas_user_id):
+        """
+        check whether the mentor is enrolled in this Canvas course
+        if enrolled, do nothing;
+        else add the mentor user to the Canvas course with Observer role
+        :param class_site_id:
+        :param mentor_canvas_user_id:
+        :return:
+        """
+        enrolled = False
+        course_section_id = ""
+        enrollments_resp = apiUtil.api_call(f"aa/CanvasReadOnly/courses/{class_site_id}/enrollments", "canvasreadonly")
+        if (enrollments_resp.text.find("errors") != -1):
+            # check for error message
+            # {"errors":[{"message":"The specified resource does not exist."}],"error_report_id":7008726002}
+            logger.warn(f"errors retrieving enrollments for class {class_site_id}: {enrollments_resp.text}")
+        else:
+            enrollments_json = json.loads(enrollments_resp.text)
+
+            for item in enrollments_json:
+                user_id=item.get("user_id")
+
+                # get the Canvas section id for student enrollment
+                if (user_id == user_canvas_user_id):
+                    course_section_id = item.get("course_section_id")
```

Comment:
I wonder if we need to return a default for this. Is it possible that the student won't be in any section or will the student always be in a section? Maybe something to log if the course_section_id is None?